### PR TITLE
added sequence API

### PR DESCRIPTION
**Pull request by nathandunn**
adds the ability to get sequences using a web-services.

- [x] check implementation
- [x] add cache
- [x] support other types
- [x] add web doc

### CODE REVIEW

**Comment by nathandunn** [comment]
See the web API doc for this.

**Comment by nathandunn** [comment]
Basically everything in the /sequence/ path.

File: grails-app/controllers/org/bbop/apollo/SequenceController.groovy [hunk 1]
```diff
@@ -213,20 +228,153 @@ class SequenceController {
...
+        sequenceService.cacheRequest(sequenceString, organismString, sequenceName, fmin, fmax,  paramMap)
+        render sequenceString
+
     }
 
+    @RestApiMethod(description = "Get sequence data as for a selected name", path = "/sequence/<organism name>/<sequence name>/<feature name>.<type>?ignoreCache=<ignoreCache>", verb = RestApiVerb.GET)
+    @RestApiParams(params = [
+            @RestApiParam(name = "organismString", type = "string", paramType = RestApiParamType.QUERY, description = "Organism common name or ID(required)")
+            , @RestApiParam(name = "sequenceName", type = "string", paramType = RestApiParamType.QUERY, description = "Sequence name(required)")
+            , @RestApiParam(name = "featureName", type = "string", paramType = RestApiParamType.QUERY, description = "If top-level feature 'id' matches, then annotate with 'selected'=1")
```
Comment (deepakunni3) [inline grails-app/controllers/org/bbop/apollo/SequenceController.groovy]: Its ambiguous if its a feature `name` or feature `uniquename`. 
Wouldn't it make sense to say `featureUniqueName` or `uniqueName` instead?

**Comment by deepakunni3** [comment]
@nathandunn Other than the comment above, everything else looks good.
Liked the idea of caching.

File: grails-app/controllers/org/bbop/apollo/SequenceController.groovy (hunk 1 above)
Comment (nathandunn) [reply grails-app/controllers/org/bbop/apollo/SequenceController.groovy]: That is because it can be either.  I'll update the doc to represent that.

**Pull request merged**